nav2_recoveries/src/clear_costmaps.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019 Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <future>
+
+#include "nav2_recoveries/clear_costmaps.hpp"
+
+namespace nav2_recoveries
+{
+
+ClearCostmaps::ClearCostmaps(rclcpp::Node::SharedPtr & node, const std::string & srv_name)
+: node_(node)
+{
+  last_clear_time_ = std::chrono::system_clock::now();
```

Comment:
I think this should be node->now(). We'd want this to use sim_time when available, IMHO. I think this would ripple out to make most of the times you use in this module to be rclcpp::Times and Durations.